trio/_util.py

```diff
@@ -177,7 +178,9 @@ def fix_one(obj):
             obj.__module__ = module_name
             if isinstance(obj, type):
                 for attr_value in obj.__dict__.values():
-                    fix_one(attr_value)
+                    # avoid infinite recursion when using typing.Generic
+                    if attr_value is not obj:
+                        fix_one(attr_value)
```

Comment:
Maybe it would be better to do more generic loop detection, by keeping a `set` of the ids of all objects we've already seen? OTOH this could wait until we actually need it...